JASP-Engine/JASP/R/commonMachineLearningClassification.R

```diff
@@ -0,0 +1,803 @@
+#
+# Copyright (C) 2019 University of Amsterdam
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+.readDataClassificationAnalyses <- function(dataset, options){
+  target                    <- NULL
+  testSetIndicator          <- NULL 
+  if(options[["target"]] != "")
+    target                  <- options[["target"]]
+  predictors                <- unlist(options['predictors'])
+  predictors                <- predictors[predictors != ""]
+  if(options[["testSetIndicatorVariable"]] != "" && options[["holdoutData"]] == "testSetIndicator")
+    testSetIndicator                  <- options[["testSetIndicatorVariable"]]
+  variables.to.read         <- c(target, predictors, testSetIndicator)
+  if (is.null(dataset)){
+    dataset <- .readDataSetToEnd(columns.as.numeric = variables.to.read, exclude.na.listwise = variables.to.read)
+  }
+  if(length(unlist(options[["predictors"]])) > 0 && options[["scaleEqualSD"]])
+    dataset[,.v(options[["predictors"]])] <- scale(dataset[,.v(options[["predictors"]])])
+  if(options[["target"]] != "")
+    dataset[, .v(options[["target"]])] <- factor(dataset[, .v(options[["target"]])])
+  return(dataset)
+}
+
+.errorHandlingClassificationAnalyses <- function(dataset, options){
+  predictors                <- unlist(options['predictors'])
+  predictors                <- predictors[predictors != ""]
+  target                    <- NULL
+  if(options[["target"]] != "")
+    target                  <- options[["target"]]
+  variables.to.read         <- c(predictors, target)
+  errors <- .hasErrors(dataset, perform, type = c('infinity', 'observations'),
+                       all.target = variables.to.read,
+                       observations.amount = "< 2",
+                       exitAnalysisIfErrors = TRUE)
+
+  dataset <- na.omit(dataset)
+  if(options[["testSetIndicatorVariable"]] != "" && options[["holdoutData"]] == "testSetIndicator" && nlevels(factor(dataset[,.v(options[["testSetIndicatorVariable"]])])) != 2){
+    JASP:::.quitAnalysis("Your test set indicator should be binary, containing only 1 (included in test set) and 0 (excluded from test set).")
+  }
+}
+
+.classificationAnalysesReady <- function(options, type){
```

Comment:
This can be done a more succinctly using `type %in% c(...)` or maybe even with S3 dispatch. Also, the function in regression is basically identical.